tools/issue-labeler/backfill.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+
+	"github.com/golang/glog"
+)
+
+type Issue struct {
+	Number int
+	Body   string
+	Labels []Label
+}
+
+type Label struct {
+	Name string
+}
+
+type IssueUpdate struct {
+	Labels []string `json:"labels"`
+}
+
+func backfill(since string, enrolledTeams map[string][]string, dryRun bool) {
+	client := &http.Client{}
+	done := false
+	page := 1
+	var issues []Issue
+	for !done {
+		url := fmt.Sprintf("https://api.github.com/repos/hashicorp/terraform-provider-google/issues?since=%s&per_page=100&page=%d", since, page)
+		req, err := http.NewRequest("GET", url, nil)
+		if err != nil {
+			glog.Exitf("Error creating request: %v", err)
+		}
+		req.Header.Add("Accept", "application/vnd.github+json")
+		req.Header.Add("Authorization", "Bearer "+os.Getenv("GITHUB_TOKEN"))
+		req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
+		resp, err := client.Do(req)
+		if err != nil {
+			glog.Exitf("Error listing issues: %v", err)
+		}
+		defer resp.Body.Close()
+		body, err := io.ReadAll(resp.Body)
+		if err != nil {
+			glog.Exitf("Error reading response body: %v", err)
+		}
+		var newIssues []Issue
+		json.Unmarshal(body, &newIssues)
+		if len(newIssues) == 0 {
+			done = true
+		} else {
+			issues = append(issues, newIssues...)
+			page++
+		}
+	}
+	for _, issue := range issues {
+		desired := make(map[string]struct{})
+		for _, existing := range issue.Labels {
+			desired[existing.Name] = struct{}{}
+		}
+		oldLength := len(desired)
+
+		for _, needed := range labels(issue.Body, enrolledTeams, false) {
+			desired[needed] = struct{}{}
+		}
+
+		if len(desired) > oldLength {
+			desiredSlice := []string{"forward/review"}
```

Comment:
I don't think we want to add `forward/review` if the ticket is already in a terminal state (`service/terraform`, `forward/linked`, or `forward/exempt`). Tickets with these labels would have already been reviewed, and they shouldn't need an additional review. If the ticket is already `forward/linked` and we are adding new service labels, the upcoming "sync" work will do the work of creating new issues internally.

Reply:
I will skip adding labels to issues that are `service/terraform` or `forward/exempt` and skip the `forward/review` label for issues that are `forward/linked`.

Reply:
I think it might actually be good to add just the service labels if we see `service/terraform` or `forward/exempt`, mainly for consistency, but wouldn't consider this critical. Context: We don't currently have a use case that requires team members to remove service labels.

Reply:
If the issue is being handled by the terraform team or exempt from forwarding shouldn't we skip adding service team labels?

Reply:
For issues being handled by the terraform team or exempt, a service label would not be changing any behavior in our systems, so I do think it is technically valid to omit them. I think my 2 hesitations are:
1. For a new issue, we will be adding the service labels before it is triaged and becomes terraform-owned or exempt, and the service labels could be left on the issue. So across all GitHub issues, service labels would be inconsistent (albeit, not in a meaningful way).
2. If we ever want a ticket to be moved out of being terraform-owned or exempt, we won't be able to trust any service labels already on the issue.

To be honest, both concerns seems a bit weak when you consider labels can be altered manually after our backfill pass, so I'm fine with leaving your implementation the way it is, but wanted to write out my thought process.